packages/plugins/transaction-logger/src/index.js

```diff
@@ -247,27 +246,44 @@ export default class TransactionLogger {
     this.embark.registerAPICall(
       'get',
       apiRoute,
-      (req, res) => {
-        res.send(JSON.stringify(this._getLogs()));
+      async (req, res) => {
+        res.send(JSON.stringify(await this._getLogs()));
       }
     );
   }
 
-  _getLogs() {
-    const data = this._readLogs();
+  async _getLogs() {
+    const data = await this._readLogs();
     return Object.values(data).reverse();
   }
 
   _saveLog(log) {
     this.writeLogFile.push(log);
   }
 
-  _readLogs() {
-    this.fs.ensureFileSync(this.logFile);
-    try {
-      return JSON.parse(this.fs.readFileSync(this.logFile));
-    } catch (_error) {
-      return {};
-    }
+  async _readLogs() {
+    return new Promise(async resolve => {
+      try {
+        await this.fs.ensureFile(this.logFile);
+        this.fs.readFile(this.logFile, (err, data) => {
```

Comment:
Since we're already in what's now an async function (`_readLogs`), I think it should be possible to:
```js
let data = (await this.fs.readFile(...)).toString()
```
If `readFile` rejects, because it's in an async function, it should be caught by your `catch`. Also, should be simpler because you don't new to manually instantiate and return a Promise. But maybe I've misunderstood something.